spec/08-pattern-matching.md

```diff
@@ -266,7 +266,7 @@ the following applies:
 
 An `unapplySeq` method in an object $x$ matches the pattern
 $x(q_1 , \ldots , q_m, p_1 , \ldots , p_n)$ if it takes exactly one argument
-and its result type is of the form `Option[($T_1 , \ldots , T_m$, Seq[S])]` (if `m = 0`, the type `Option[Seq[S]]` is also accepted).
+and its result type is of the form `Option[($T_1 , \ldots , T_m$, scala.collection.Seq[S])]` (if `m = 0`, the type `Option[scala.collection.Seq[S]]` is also accepted).
```

Comment:
I agree with @jrudolph in https://github.com/scala/scala-dev/issues/591#issuecomment-458895442, `_*` and `unapplySeq` should return Seqs, not `scala.collection.Seq`, which probably requires `Seq.unapply` to accept `scala.collection.Seq`s.

Reply:
Thanks to name-based pattern matching you can already use different return types for `unapplySeq`. They are only converted to `scala.collection.Seq` (previously `scala.Seq`) when you match with `@ _*`. This is the dual of varargs, using the same `_*` syntax. It would be very inconsistent to give it a different type.

Note that this is unrelated to actually matching a Seq type (`case Seq(x, y, z) => ...`).

Reply:
One of the reasons for moving varargs back to `scala.collection.Seq` is performance.  Is that relevant for `..., xs: _*)` unapply too?